core/rawdb/database.go

```diff
@@ -451,7 +451,14 @@ func (s *stat) Count() string {
 
 // InspectDatabase traverses the entire database and checks the size
 // of all different categories of data.
-func InspectDatabase(db ethdb.Database, keyPrefix, keyStart []byte) error {
+func InspectDatabase(db ethdb.Database, keyPrefix, keyStart []byte, options ...InspectDatabaseOption) error {
```

Comment:
These changes can be [reduced to just a few lines](https://github.com/ava-labs/libevm/commit/56700320ed60d8bc3f4fa2a6a240ad772e963359), which will reduce the chance of a merge conflict when doing an upstream sync. By abstracting the functionality it also directly communicates the intent of the modifications too.

Although the commit doesn't include the exact functionality you've got here, it provides the necessary options to recreate it externally in full generality.